### DISCUSSION TITLE
Computer spec for OpenSAFELY

### DISCUSSION THREAD

**Discussion by eiliskeeble** [opensafely/documentation#971] [Installing OpenSAFELY]
Question from IT support at my organisation: Is there any information on the computer specification needed to run OpenSAFELY well? 

I used it on a personal computer (2012 macbook) initially where it ran well but have since transitioned to a work computer (windows hp probook) where things take a lot longer to run and there seems to be more of a pause before analysis starts to run. 

Is it a case of just seeing what each software requires individually to work well or is there some general guidance from experiences with OpenSAFELY so far?

**Comment by iaindillingham** [comment]
Hi @eiliskeeble, thanks for your question.

How much time does it take your work computer to run the study? And your personal computer? Is the study the same, in each case? We haven't written up any guidance on system requirements, although you may wish to consider allocating more memory/CPU to Docker.

I've run studies on 2013 and 2020 MacBook Pros; the former took longer than the latter, but not unreasonably longer (seconds/minutes rather than minutes/hours).

The time it takes to run a study will depend on the nature of the study. However, there is a section in the documentation called [*Writing performant code*](https://docs.opensafely.org/memory-efficient-working/), which you may find useful and interesting.
